dask/array/tests/test_gufunc.py

```diff
@@ -615,3 +616,18 @@ def stats(x):
 
     assert_eq(sum, sum)
     assert_eq(mean, mean)
+
+
+def test_apply_gufunc_with_meta():
+    def stats(x):
+        return np.mean(x, axis=-1), np.std(x, axis=-1)
+
+    a = da.random.normal(size=(10, 20, 30), chunks=(5, 5, 30))
+    meta = (np.ones(0, dtype=np.float64), np.ones(0, dtype=np.float32))
+    result = apply_gufunc(stats, "(i)->(),()", a, meta=meta)
+    mean, std = result
+    assert isinstance(result, tuple)
+    assert mean.compute().shape == (10, 20)
+    assert std.compute().shape == (10, 20)
+    for expected, actual in zip(meta, result):
+        assert expected.dtype == actual._meta.dtype
```

Comment:
Maybe we could do something like 

```python
expected = stats(a.compute())
assert_eq(expected[0], result[0])
assert_eq(expected[1], result[1])
```

Using assert_eq checks lots of things, like shape, dtype, and other metadata.  It's generally good testing hygiene in Dask to use it when possible.

Reply:
Thanks for the feedback! I switched to using `assert_eq` as suggested, which actually exposed the fact that passing `np.float32` as the dtype for the second meta was not actually surviving after calling `compute`. I had to pass in the dtype in the gufunc and in the meta for it to actually take effect. The test illustrates that now, but let me know if you think it makes sense to add more tests to illustrate the usage.